### PR TITLE
5413 update dicefocal include foreground

### LINKED ISSUE #5413: include_background=False, softmax=True doesn't work properly in DiceFocalLoss for binary segmentation
**Environment**
monai 1.0.1

**Describe the bug**
When using `include_background=False, softmax=True` in DiceFocalLoss for binary segmentation, the Dice part produces this warning and does not apply softmax when calculating the Dice loss:
```
UserWarning: single channel prediction, `softmax=True` ignored.
  warnings.warn("single channel prediction, `softmax=True` ignored.")
```

**To Reproduce**
```
import torch
from monai.losses import DiceFocalLoss
from monai.networks import one_hot

tx = torch.rand(1, 2, 256, 256) # logits
ty = one_hot((torch.rand(1, 1, 256, 256) > 0.5).long(), 2) # one-hot label
loss = DiceFocalLoss(include_background=False, softmax=True)
loss(tx, ty)
```

**Expected behavior & Additional context**
By `include_background=False`, since the first channel is removed before passing the input and target to Dice and Focal losses, the Dice loss cannot apply softmax to the input which now has only one channel.
https://github.com/Project-MONAI/MONAI/blob/f407fcce1c32c050f327b26a16b7f7bc8a01f593/monai/losses/dice.py#L849-L860

### LINKED ISSUE #5431: tests for torch 1.13 (and drop support for torch 1.7.x?)
**Is your feature request related to a problem? Please describe.**
- add tests for the new pytorch releases
- drop the tests for v1.7.x?

### PR DESCRIPTION
**Pull request by wyli**
Signed-off-by: Wenqi Li <wenqil@nvidia.com>

Fixes #5413

### Description
excluding background shouldn't be done before softmax

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [x] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests  --disttests`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

**Review by KumoLiu — APPROVED** [review]
Thanks for updating, it looks good to me!
